go/vt/mysqlctl/xtrabackupengine.go

```diff
@@ -554,16 +557,28 @@ func (be *XtrabackupEngine) extractFiles(ctx context.Context, logger logutil.Log
 		// Create the decompressor if needed.
 		if compressed {
 			var decompressor io.ReadCloser
-
+			var deCompressionEngine = bm.CompressionEngine
+			if deCompressionEngine == "" {
+				// For backward compatibility. Incase if Manifest is from N-1 binary
+				// then we assign the default value of compressionEngine.
+				deCompressionEngine = PgzipCompressor
```

Comment:
Is it possible that a user would have compressed with a non-default engine, and that this wouldn't have been recorded in the manifest file?

If yes, then in that case what error would the user get?

Wondering if it would be more helpful to require that the user specify compression engine if it's not present in the manifest, even nicer if we can hint what we think the right value is (e.g. if we can peek at magic numbers).

Reply:
Is it possible that a user would have compressed with a non-default engine, and that this wouldn't have been recorded in the manifest file?
>> No, thats not possible .. current binary in production uses pgzip as default. Only scenario when we don't see a value in manifest is during our first deployment to this feature. Once this feature is deployed then we expect manifest to alway have some value.

If we end up reverting the feature then what you specify can happen but even in this case "a flag user can use to specify compression engine" won't help as we will be reverting to older bits. The solution I can think of in that case is to move values to default (pgzip), perform a manual backup and then revert the binary.

Reply:
> Only scenario when we don't see a value in manifest is during our first deployment to this feature.

In that scenario, I guess they will see an error like "gzip can't decompress file". I think it would be helpful if the user got a recommendation to remember to set `--external-decompressor`, or if we told the user that they are required to set `--external-decompressor` when the manifest doesn't specify compression.

Just my opinion on UX, so feel free to move on if you disagree.